### PR TITLE
Disallow task inputs/outputs chaining

### LINKED ISSUE #6281: Remove LenientTaskInputsDeprecationSupport

### PR DESCRIPTION
**Pull request by wolfs**
Fixes #6281.

### CODE REVIEW

File: buildSrc/subprojects/binary-compatibility/src/main/groovy/org/gradle/binarycompatibility/rules/AbstractGradleViolationRule.groovy [hunk 1]
```diff
@@ -128,7 +132,7 @@ abstract class AbstractGradleViolationRule extends AbstractContextAwareViolation
                 return sourceFile
             }
         }
-        throw new RuntimeException("No source file found for: $className")
+        return null
```
Comment (lptr) [inline buildSrc/subprojects/binary-compatibility/src/main/groovy/org/gradle/binarycompatibility/rules/AbstractGradleViolationRule.groovy]: Why this change as part of this PR?

Comment (wolfs) [reply buildSrc/subprojects/binary-compatibility/src/main/groovy/org/gradle/binarycompatibility/rules/AbstractGradleViolationRule.groovy]: If we don't have this change, then we get an NPE when running the binary compatibility check: The class `CompatibilityAdapterForTaskInputs` on the public API has been removed, so we try to get the source code for it and file with an exception if we don't find it. I guess the check could be more clever, like only trying to get the source file if the change is not a removal. WDYT?

Comment (lptr) [reply buildSrc/subprojects/binary-compatibility/src/main/groovy/org/gradle/binarycompatibility/rules/AbstractGradleViolationRule.groovy]: Sounds like doing so would make the use case more explicit in the code, too. I'd say let's go for it.

Comment (wolfs) [reply buildSrc/subprojects/binary-compatibility/src/main/groovy/org/gradle/binarycompatibility/rules/AbstractGradleViolationRule.groovy]: Done.

**Comment by lptr** [comment]
LGTM
